src/core/qgssnappingutils.cpp

```diff
@@ -615,7 +615,7 @@ void QgsSnappingUtils::readConfigFromProject()
        layerIdList.size() != snapToList.size() )
     return;
 
-  if ( !snappingDefinedInProject )
+  if ( layerIdList.isEmpty() || !snappingDefinedInProject || !ok )
```

Comment:
I do not see any harm here, but I fail to understand what does that change fix...?